### ISSUE TITLE
Error on overload on self

### ISSUE THREAD

**Issue by tkukushkin** [microsoft/pyright#1829]
**To Reproduce**
Minimal example:
```python
from __future__ import annotations
from typing import Any, Generic, TypeVar, overload, Optional

T = TypeVar('T')

class Foo(Generic[T]):
    @overload
    def __call__(self: Foo[None]) -> None:
        ...
        
    @overload
    def __call__(self, value: T) -> None:
        ...
        
    def __call__(self, value: Any = None) -> None:
        ...
```
Result:
```
.../test.py
.../test.py:15:9 - error: Overloaded function implementation is not consistent with signature of overload 1
    Type "(self: Foo[T@Foo], value: Any | None = None) -> None" cannot be assigned to type "(self: Foo[None]) -> None"
      Parameter 1: type "Foo[None]" cannot be assigned to type "Foo[T@Foo]"
        TypeVar "T@Foo" is invariant
          Type "None" cannot be assigned to type "object" (reportGeneralTypeIssues)
```

**Expected behavior**
No errors produced. It [works](https://mypy-play.net/?mypy=latest&python=3.9&gist=0c4196b99ac264b70c77ecfa0d9d7cbc) in mypy (but not with the `__call__` because of another bug).

**VS Code extension or command-line**
command line, 1.1.137

**Comment by tkukushkin** [comment]
Such implementation works ok:
```python
def __call__(self: Foo[Any], value: Any = None) -> None:
    ...
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.139, which I've just published. It will also be included in the next release of pylance.
